packages/app-admin/src/components/IconPicker/config/IconPackProvider.tsx

```diff
@@ -1,64 +1,46 @@
-import React, { useCallback, useState, useRef, useEffect } from "react";
+import React, { useCallback } from "react";
 import { Property, useIdGenerator } from "@webiny/react-properties";
 
-export interface IconProps {
-    type: string;
+type Icon = {
+    type: "icon";
     name: string;
-    skinToneSupport?: boolean;
-    category?: string;
     value: string;
+    category?: string;
     width?: number;
-}
+};
 
-export type IconPackProviderProps = {
+export type Emoji = {
+    type: "emoji";
     name: string;
-    provider: () => Promise<IconProps[]> | IconProps[];
+    value: string;
+    category: string;
+    skinToneSupport: boolean;
 };
 
-export const IconPackProvider = ({ name, provider }: IconPackProviderProps) => {
-    const getId = useIdGenerator("iconPackProvider");
-
-    const isMounted = useRef(true);
-    const [icons, setIcons] = useState<IconProps[]>([]);
-    const [isLoading, setIsLoading] = useState(false);
-    const [isInitialized, setIsInitialized] = useState(false);
+type Custom = {
+    type: "custom";
+    name: string;
+    value: string;
+};
 
-    const initialize = useCallback(async () => {
-        setIsLoading(true);
+export type ProviderIcon = Icon | Emoji | Custom;
 
-        const iconsData = await provider();
+export type IconPackProviderProps = {
+    name: string;
+    provider: () => Promise<ProviderIcon[]> | ProviderIcon[];
+};
 
-        if (!isMounted.current) {
-            return;
-        }
+export const IconPackProvider = ({ name, provider }: IconPackProviderProps) => {
+    const getId = useIdGenerator("iconPackProvider");
 
-        setIcons(iconsData);
-        setIsInitialized(true);
-        setIsLoading(false);
+    const load = useCallback(async () => {
+        return await provider();
     }, [provider]);
```

Comment:
Not sure why you need this, since the `provider` is already a closure, its reference will never change, and so it is safe to just pass to the `<Property>`.